rpc/wrtc_signaling_answerer.go

```diff
@@ -238,6 +246,23 @@ func (ans *webrtcSignalingAnswerer) Stop() {
 // the designated WebRTC data channel is passed off to the underlying Server which
 // is then used as the server end of a gRPC connection.
 func (ans *webrtcSignalingAnswerer) answer(client webrtcpb.SignalingService_AnswerClient) (err error) {
+	// Maintain and eventually log a collection of stats for each answering attempt.
+	stats := &answererStats{}
+	defer func() {
+		if ans.logStats {
+			statsJSON, err := json.Marshal(stats)
+			if err != nil {
+				ans.logger.Errorf("Error converting answerer stats to JSON: %w", err)
+				return
+			}
+			if stats.success {
+				ans.logger.Infow("Connection establishment success", "stats", string(statsJSON))
```

Comment:
This log looks like:
```
2024-04-24T14:17:49.058-0400	INFO	robot_server.external_signaler	rpc/wrtc_signaling_answerer.go:259	Connection establishment success	{"stats": "{\"answer_request_init_received\":\"2024-04-24T14:17:48.80109-04:00\",\"num_answer_updates\":3,\"average_answer_update\":29236,\"max_answer_update\":49500,\"local_ice_candidates\":[{\"gathered_at\":\"2024-04-24T14:17:48.803465-04:00\",\"candidate\":\"udp4 host 127.0.0.1:57413\"},{\"gathered_at\":\"2024-04-24T14:17:48.803721-04:00\",\"candidate\":\"udp4 host 192.168.1.127:50619\"},{\"gathered_at\":\"2024-04-24T14:17:48.861006-04:00\",\"candidate\":\"udp4 srflx 66.24.146.163:61077 related 0.0.0.0:61077\"}],\"remote_ice_candidates\":[{\"received_at\":\"2024-04-24T14:17:48.987992-04:00\",\"candidate\":\"candidate:392662914 1 udp 2130706431 859ad36c-1539-485d-bb53-046e3321f263.local 63388 typ host\"},{\"received_at\":\"2024-04-24T14:17:48.995173-04:00\",\"candidate\":\"candidate:1938624796 1 udp 1694498815 66.24.146.163 57236 typ srflx raddr 0.0.0.0 rport 57236\"},{\"received_at\":\"2024-04-24T14:17:48.997028-04:00\",\"candidate\":\"candidate:1938624796 1 udp 1694498815 66.24.146.163 58063 typ srflx raddr 0.0.0.0 rport 58063\"},{\"received_at\":\"2024-04-24T14:17:48.997058-04:00\",\"candidate\":\"candidate:2838388448 1 udp 16777215 34.203.251.220 28383 typ relay raddr 0.0.0.0 rport 64553\"},{\"received_at\":\"2024-04-24T14:17:48.997071-04:00\",\"candidate\":\"candidate:392662914 1 udp 2130706431 859ad36c-1539-485d-bb53-046e3321f263.local 59069 typ host\"}]}"}
```

Reply:
can we make it so the fields of the blob is spread (so like `ans.logger.Infow("Connection establishment success", "answer_request_init_received", xxxx, "num_answer_updates", x)

tho I don't feel that strongly about it. I think it might be slightly easier to read and parse for upstream, but not by much.

Reply:
Yeah it's interesting; logging the stringified JSON blob here is pretty terrible in a readability sense. But, if we start to use Go structured logging with `Infow`, there are a couple issues:
- https://viam.atlassian.net/browse/APP-4143 means many of these fields won't show up in the `Logs` tab and we'll have to look at the `logs` DB for the time being to see the real values
- There will be no real Go "structure" representing the stats, only key-value pairs that are potentially more subject to change

So, I'm torn. Neither solution is really optimal IMO, but maybe this JSON blob is acceptable for the time being until we potentially start sending these metrics to some metric collector and stop logging them here? To be clear, a log similar to the one I've copied above will show up in RDK logs every time the robot succeeds or fails a connection establishment with a client. Do we think users will find that troubling/annoying?